File: web/src/dtos/Area.ts
import Common from "../utility/Common";
import { CityRes } from "./City";
import { PagedReq } from "./PagedReq";

export interface AreaRes {
  areaId?: number;
  name?: string;
  code?: string;
  cityId?: number;

  city?: CityRes;
}

export class AreaReqEdit {
  name?: string;
  code?: string;
  cityId?: number;

  constructor(cityId?: number) {
    this.cityId = cityId;
  }
}

export class AreaReqSearch extends PagedReq {
  cityId?: number;

  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    cityId = 0
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.cityId = cityId;
  }
}

File: api/Dtos/Area.cs
using api.Utility.Paging;
using System.ComponentModel.DataAnnotations;

namespace api.Dtos
{
    public class AreaRes
    {
        public int AreaId { get; set; }
        public string? Name { get; set; }
        public string? Code { get; set; }

        // Foreign keys
        public int CityId { get; set; }
        public CityRes? City { get; set; }
    }

    public class AreaReqEdit
    {
        [Required, MaxLength(255)]
        public string? Name { get; set; }
        [Required, MaxLength(50)]
        public string? Code { get; set; }

        // Foreign keys
        public int CityId { get; set; }
    }

    public class AreaReqSearch : PagedRequestDto
    {
        public int CityId { get; set; }
    }
}
